boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_shared.dts

```diff
@@ -0,0 +1,133 @@
+/ {
+leds:	leds {
+		compatible = "gpio-leds";
+		rgb1_red: led_0 {
+			gpios = <&gpio0 7 GPIO_ACTIVE_HIGH>;
+			label = "0 LED_RGB_RED CORE_APP";
+		};
+		rgb1_green: led_1 {
+			gpios = <&gpio0 25 GPIO_ACTIVE_HIGH>;
+			label = "0 LED_RGB_GREEN CORE_APP";
+		};
+		rgb1_blue: led_2 {
+			gpios = <&gpio0 26 GPIO_ACTIVE_HIGH>;
+			label = "0 LED_RGB_BLUE CORE_APP";
+		};
+		rgb2_red: led_3 {
+			gpios = <&gpio0 28 GPIO_ACTIVE_HIGH>;
+			label = "1 LED_RGB_RED CORE_NET";
+		};
+		rgb2_green: led_4 {
+			gpios = <&gpio0 29 GPIO_ACTIVE_HIGH>;
+			label = "1 LED_RGB_GREEN CORE_NET";
+		};
+		rgb2_blue: led_5 {
+			gpios = <&gpio0 30 GPIO_ACTIVE_HIGH>;
+			label = "1 LED_RGB_BLUE CORE_NET";
+		};
+		led1_blue: led_6 {
+			gpios = <&gpio0 31 GPIO_ACTIVE_HIGH>;
+			label = "2 LED_MONO_BLUE CORE_APP";
+		};
+		led2_green: led_7 {
+			gpios = <&gpio1 0 GPIO_ACTIVE_HIGH>;
+			label = "3 LED_MONO_GREEN CORE_APP";
+		};
+		led3_green: led_8 {
+			gpios = <&gpio1 1 GPIO_ACTIVE_HIGH>;
+			label = "4 LED_MONO_GREEN CORE_APP";
+		};
+
+	};
+
+	buttons {
+		compatible = "gpio-keys";
+		button_1_vol_dn: button_1_vol_dn {
+			gpios = <&gpio0 2 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+			label = "Push button 1";
+		};
+		button_2_vol_up: button_2_vol_up {
+			gpios = <&gpio0 3 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+			label = "Push button 2";
+		};
+		button3: button_3 {
+			gpios = <&gpio0 4 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+			label = "Push button 3";
+		};
+		button4: button_4 {
+			gpios = <&gpio0 6 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+			label = "Push button 4";
+		};
+		button5: button_5 {
+			gpios = <&gpio0 5 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+			label = "Push button 5";
+		};
+	};
+
+	nrf5340_audio_dk {
+		compatible = "gpio-keys";
+		hw_codec_gpio_in: hw_codec_gpio_in {
+			gpios = <&gpio0 20 GPIO_ACTIVE_LOW>;
+			label = "GPIO from HW codec";
+		};
+
+		hw_codec_irq_in: hw_codec_irq_in {
+			gpios = <&gpio0 19 GPIO_ACTIVE_LOW>;
+			label = "Interrupt from HW codec";
+		};
+
+		hw_codec_reset_out: hw_codec_reset_out {
+			gpios = <&gpio0 18 GPIO_ACTIVE_LOW>;
+			label = "Reset HW codec";
+		};
+
+		hw_codec_sel_out: hw_codec_sel_out {
+			gpios = <&gpio0 21 GPIO_ACTIVE_LOW>;
+			label = "HW codec selector. Low selects OB HW codec, high \
+				 selects external HW codec";
+		};
+
+		spi_sel_in: spi_sel_in {
+			gpios = <&gpio0 22 GPIO_ACTIVE_LOW>;
+			label = "SPI select in. High = nRF SPI to HW codec. \
+				 Low = FTDI to HW codec";
+		};
+
+		pmic_iset_out: pmic_iset_out {
+			gpios = <&gpio0 23 GPIO_ACTIVE_HIGH>;
+			label = "PMIC ISET";
+		};
+
+		board_id_en_out: board_id_en_out {
+			gpios = <&gpio0 24 GPIO_ACTIVE_HIGH>;
+			label = "Board revision readback enable";
+		};
+
+		board_id_in: board_id_in {
+			gpios = <&gpio0 27 GPIO_ACTIVE_HIGH>;
+			label = "Board revision analog readback";
+		};
+
+		curr_mon_alert_in: curr_mon_alert_in {
+			gpios = <&gpio1 15 GPIO_ACTIVE_LOW>;
+			label = "Curr mon alert in from INA231";
```

Comment:
this looks wrong, why a driver specific GPIO under `gpio-keys`?

Reply:
We have a task to correct this in NCS, but we need that resolved first before this can be removed.

Reply:
this cannot get in as-is.

Reply:
We are working on changing out the entire INA231 driver to the upstream one, but this will take some time and it would have been great to get this merged despite of the ongoing efforts on that front. Do you have any suggestions for a temporary solution?